### ISSUE TITLE
Issues with the choices library

### ISSUE THREAD

**Issue by MotionPhix** [protonemedia/laravel-splade#96]
I have come to a conclusion that the choices library is causing some issues to laravel splade especially the x-splade-select component most notably. I have been working on a project and have spent hours trying to resolve it until I just decided to remove the choices attribute from the x-splade-select component that things finally worked out. Take a look below:

![normal_table](https://user-images.githubusercontent.com/59722054/198865765-cbc38cb5-fc30-4a7d-91ca-bc3fbb94b5f7.png)

That's my data in just a normal table.

![choices_disabled__](https://user-images.githubusercontent.com/59722054/198865778-c391994b-001d-4a01-9e32-ec2e4c5114a6.png)

This is my code with choices attribute disabled. When I click on the edit icon in the table, I get the following screen

![data_shown_when_choices_is_off](https://user-images.githubusercontent.com/59722054/198865814-e7e4f879-586b-4e00-a73c-e27b90ef8132.png)

Notice how on the **Company** the name appears correctly matching the name in the table under **Organisations**? Now, take a look if I enable choices attribute.

![choices_active](https://user-images.githubusercontent.com/59722054/198865887-f0a526a3-8fff-41c1-affc-d660dca19b6b.png)

This is my code with choices attribute included. Now when I click on edit in the same table, I get the following.

![deffects_choices](https://user-images.githubusercontent.com/59722054/198865918-b19c2b52-b593-4d7e-ae31-b7e22e550fb8.png)

See how the name under **Company** doesn't match the name under **Organisations** in the table? It's because of the choices library. I don't know why, though. Another issue with the choices library that I noticed is that when you put **searchEnabled => false** and then you click in the select field, and then clicking away without selecting any item, the selection list won't close no matter what you do. Then only way to close it is if you actually make a selection. Would appreciate if you could take a look into those issues.

**Comment by pascalbaljet** [comment]
I can't reproduce the first problem. I've added a test for it as well: https://github.com/protonemedia/laravel-splade/commit/5adb94de81d86df63d1bea198778812d3c319558

What version of Splade (JS and PHP) are you using?

I'll look into the other issue as well :)

**Comment by pascalbaljet** [comment]
The second issue has been fixed in v0.6.13, and was related to #74.

**Comment by MotionPhix** [comment]
> I can't reproduce the first problem. I've added a test for it as well: [5adb94d](https://github.com/protonemedia/laravel-splade/commit/5adb94de81d86df63d1bea198778812d3c319558)
> 
> What version of Splade (JS and PHP) are you using?
> 
> I'll look into the other issue as well :)

I have committed the project to [github](https://github.com/MotionPhix/splade-fan) so you can use that to see what's happening. By the way, I updated splade 0.6.13 and laravel is 9.19. Please take a look at the files. Thank you

**Comment by MotionPhix** [comment]
> The second issue has been fixed in v0.6.13, and was related to #74.

On a separate note, I went back to the docs to read more about the **x-splade-select** component. You see, the example data you are using which is for a very simple list of countries is not probably the closest to real-world scenarios we will have, because in real world we will have an array of arrays. Something like this:

`

$countries = [

[
'id' => 1,
'country_code' => 'be',
'name' => 'Belgium',
],

[
'id' => 2,
'country_code' => 'nl',
'name' => 'Netherlands
]

]

`

Now, use data like that to work on the component and observe what happens. Thank you

**Comment by pascalbaljet** [comment]
`Arr::pluck($countries, 'name', 'country_code')` 😉

**Comment by MotionPhix** [comment]
> `Arr::pluck($countries, 'name', 'country_code')` 😉

Will this work with an eloquent model collection?

**Comment by pascalbaljet** [comment]
The `Collection` class has a `pluck` method as well:

```php
User::get()->pluck('name', 'id');
```

For remote sources (still WIP), I've added support for defining the label and value key, maybe I'll bring it over so you could do something like this:

```blade
<x-splade-select :options="$users" option-label="name" option-value="id" />
```

**Comment by MotionPhix** [comment]
Yes, that will be really great

**Comment by pascalbaljet** [comment]
The `option-label` and `option-value` attributes are available in v0.7 👌

**Comment by pascalbaljet** [comment]
> I have pushed the project to [github](https://github.com/MotionPhix/splade-fan) so you can use that to see what's happening. By the way, I updated splade 0.6.13 and laravel is 9.19. Please take a look at the files. Thank you

Can confirm. I'm working on a fix.

**Comment by pascalbaljet** [comment]
Fixed in v0.7.3.
